src/fluree/json_ld/impl/iri.cljc

```diff
@@ -45,3 +46,16 @@
                              {:status 400
                               :error  :json-ld/invalid-iri}
                              e)))))
+
+(defn join
+  "Returns an IRI string for string s relative to base IRI."
+  [base ^String s]
+  #?(:clj  (->> s (.resolve (URI/create base)) .toString)
+     :cljs (-> s (js/URL. base) .toString)))
+
+(defn absolute?
+  "Returns true if given IRI is absolute, false otherwise.
+  NB: A false return value does NOT imply that it is a valid relative IRI."
+  [iri]
+  #?(:clj  (try (-> iri URI/create .isAbsolute) (catch Exception _ false))
+     :cljs (boolean (try (js/URL. iri) (catch :default _ false)))))
```

Comment:
I'm not sure JavaScript's `URL` is designed to support IRIs? Even if a JS engine does, it doesn't mean they all will.

I'd instead prefer a super simplified version and avoid pulling out the conditionals. I think `(str/includes? iri ":")` would be sufficient. We don't need to support things like `../xyz` here, so your `join` function can just be `(str base iri)`. There is no need to check for trailing/starting `/`, `#`, etc. before joining.... some responsibility does lie with the `@context` creator to do things properly.

Alternatively the json-ld reference implementation uses this regex for "absolute" checks... slightly more robust, which I wouldn't think is necessary and will be a little bit slower, but perhaps best to use just for consistency: 
https://github.com/digitalbazaar/jsonld.js/blob/1f34863d0ba9e0fc631beea084da63f3e6c8eaf6/lib/url.js#L280C11-L280C11

Reply:
On second thought, I think the reference implementation isn't ideal as it doesn't support international characters. I personally prefer `(str/includes? iri ":")`, but only for testing if `@vocab` is relative or not, not necessarily as a universal `absolute?` test - but we only need to know here if @vocab is relative, not that it is a full proper independent IRI.

Reply:
Done in https://github.com/fluree/json-ld/pull/23/commits/16059ca81354c0a4729af4f57cf35b8cbbf38650